sway-ast/Cargo.toml

```diff
@@ -10,6 +10,7 @@ description = "Sway's parser"
 
 [dependencies]
 extension-trait = "1.0.1"
-num-bigint = "0.4.3"
+num-bigint = { version = "0.4.3", features = ["serde"] }
 num-traits = "0.2.14"
+serde = { version = "1.0", features = ["derive"] }
```

Comment:
Hmm I wonder if we should have a `serde` feature that enables this just like most of the crates we depend on. The serde feature would enable the actual serialization implementations. 

I am imagining serialization is not needed by the compiler itself but it is a good option to offer to downstream crates (for example lsp).

Reply:
Compiler could have a parse cache, no?

Reply:
Hmm that makes sense 👌